test/base/test_batch.py

```diff
@@ -38,15 +37,6 @@ def test_batch():
     assert 'a' not in b
     with pytest.raises(AssertionError):
         Batch({1: 2})
-    assert Batch(a=[np.zeros((2, 3)), np.zeros((3, 3))]).a.dtype == object
-    with pytest.raises(TypeError):
-        Batch(a=[np.zeros((3, 2)), np.zeros((3, 3))])
-    with pytest.raises(TypeError):
-        Batch(a=[torch.zeros((2, 3)), torch.zeros((3, 3))])
-    with pytest.raises(TypeError):
-        Batch(a=[torch.zeros((3, 3)), np.zeros((3, 3))])
-    with pytest.raises(TypeError):
-        Batch(a=[1, np.zeros((3, 3)), torch.zeros((3, 3))])
```

Comment:
why delete this?

Reply:
Because these operations are not supported yet by the Batch class, and it comes down basically to wanting to construct a numpy array with numpy arrays of different shapes which is not supported by numpy _"At the core of the NumPy package, is the ndarray object. This encapsulates n-dimensional arrays of homogeneous data types"._ I don't know If these kind of operations will ever be supported by Batch but right now they are not. Maybe you would like to just comment it out and put a TODO note there if you think that this will supported in the future.

Reply:
I have reverted it back, I changed whenever `np.array([a1, a2])` is called on `a1` and `a2` of different shapes to `np.array([a1, a2], dtype=object)`.